engine/h2shared/quakefs.c

```diff
@@ -313,15 +315,25 @@ static pack_t *FS_LoadPackFile (const char *packfile, int paknum, qboolean base_
 		gameflags |= check_known_paks (paknum, numpackfiles, crc);
 	else	gameflags |= GAME_MODIFIED;
 
+	pack = (pack_t *) Z_Malloc (sizeof(pack_t), Z_MAINZONE);
+	/* get the hash table size from the number of files in the pak */
+	for (i = 1; i < MAX_FILES_IN_PACK; i <<= 1)
```

Comment:
Is this `i <<= 1` truly correct? What will happen with `1 << 2047`???

Reply:
Oh.. You are stopping `if (i > numpackfiles)` - so this is a way of finding the proper power-of-two number.

Reply:
Yes, it just to rounds up numpakfiles to the next power of 2. It will also never exceed MAX_FILES_IN_PACK, so even packs with the maximum amount of files are safe to use.